test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -92,9 +93,94 @@ session_simulator::set_read_timestamp(uint64_t ts)
     _read_ts = ts;
 }
 
+int
+session_simulator::decode_timestamp_config_map(std::map<std::string, std::string> &config_map,
+  uint64_t &commit_ts, uint64_t &durable_ts, uint64_t &prepare_ts, uint64_t &read_ts)
+{
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    auto pos = config_map.find("commit_timestamp");
+    if (pos != config_map.end()) {
+        commit_ts = ts_manager->hex_to_decimal(pos->second);
+        if (commit_ts == 0) 
+            WT_SIM_RET_MSG(EINVAL, "Illegal commit timestamp: zero not permitted.");
+        config_map.erase(pos);
+    }
+
+    pos = config_map.find("durable_timestamp");
+    if (pos != config_map.end()) {
+        durable_ts = ts_manager->hex_to_decimal(pos->second);
+        if (durable_ts == 0) 
+            WT_SIM_RET_MSG(EINVAL, "Illegal durable timestamp: zero not permitted.");
+        config_map.erase(pos);
+    }
+
+    pos = config_map.find("prepare_timestamp");
+    if (pos != config_map.end()) {
+        prepare_ts = ts_manager->hex_to_decimal(pos->second);
+        if (prepare_ts == 0) 
+            WT_SIM_RET_MSG(EINVAL, "Illegal prepare timestamp: zero not permitted.");
+        config_map.erase(pos);
+    }
+
+    pos = config_map.find("read_timestamp");
+    if (pos != config_map.end()) {
+        read_ts = ts_manager->hex_to_decimal(pos->second);
+        if (read_ts == 0) 
+            WT_SIM_RET_MSG(EINVAL, "Illegal read timestamp: zero not permitted.");
+        config_map.erase(pos);
+    }
+
+    return (config_map.empty() ? 0 : EINVAL);
+}
+
+int
+session_simulator::timestamp_transaction(const std::string &config)
+{
+    /* Make sure that the transaction from this session is running. */
+    assert(_txn_running);
+
+    /* If no timestamp was supplied, there's nothing to do. */
+    if (config.empty())
+        return (0);
+
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    std::map<std::string, std::string> config_map;
+
+    ts_manager->parse_config(config, config_map);
+
+    uint64_t commit_ts = 0, durable_ts = 0, prepare_ts = 0, read_ts = 0;
+
+    /* Decode a configuration string that may contain multiple timestamps and store them here. */
+    WT_SIM_RET_MSG(
+      decode_timestamp_config_map(config_map, commit_ts, durable_ts, prepare_ts, read_ts),
+      "Incorrect config passed to 'timestamp_transaction': '" + config + "'");
+
+    /* Check if the timestamps were included in the configuration string and set them. */
+    if (commit_ts != 0) {
+        set_commit_timestamp(commit_ts);
```

Comment:
I don't think we need to check for the `0` value? Could we do this:
```suggestion
        set_commit_timestamp(commit_ts);
        set_durable_timestamp(durable_ts);
        set_prepare_timestamp(prepare_ts);
        set_read_timestamp(read_ts);
```

Reply:
The check for 0 is there since all the timestamps are initialised to 0 before calling `decode_timestamp_config_map`. Then if the timestamp is not specified in the configuration string `decode_timestamp_config_map` won't update it and it will remain 0 which means it should not be set.

Reply:
I think it decreases the readability to have all those checks against 0. Calling the function with a ts of `0` will just set the value again. @bvpvamsikrishna ,what do you think?

Reply:
But if the timestamp has been previously set, we don't wan tot reset it to 0.

Reply:
If we don't check against '0', there will be a problem with logic. Another option is instead of assigning '0' we can read the values from the session and assign them with old values so that even if the values are not passed, reassigning them with the same values will not be a problem. 
I am ok, with either approach.

Reply:
I understand, it is fine the way it is.